jni_bindings/src/lib.rs

```diff
@@ -457,273 +536,268 @@ pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node
-            JObject::null().into()
-        }
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Attribute_drop(
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Attribute_drop(
     mut env: JNIEnv,
     _: JClass,
+    // non-null raw pointer to Attribute
     this: jlong,
 ) {
-    let result = catch_unwind(|| unsafe {
-        let dom: *mut Attribute = jlong_to_pointer::<Attribute>(this).as_mut().unwrap();
-        let dom = Box::from_raw(dom);
-        drop(dom);
-    });
-
-    if let Err(err) = result {
-        let exception_class = env.find_class("java/lang/RuntimeException").unwrap();
-        let message = format!("Rust panic occurred in Attribute drop: {:?}", err);
-        env.throw_new(exception_class, message).unwrap();
+    let this = this as *mut Attribute;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Attribute::drop called with `this` as null pointer",
+        )
+        .unwrap();
+    } else {
+        let _attr = Box::from_raw(this);
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Attribute_get_1name<'local>(
-    env: JNIEnv<'local>,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Attribute_get_1name<'local>(
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    attr: jlong,
+    // non-null raw pointer to Attribute
+    this: jlong,
 ) -> JString<'local> {
-    let mut clone = unsafe { env.unsafe_clone() };
-    let result = catch_unwind(|| {
-        let attribute: &Attribute =
-            unsafe { jlong_to_pointer::<Attribute>(attr).as_mut().unwrap() };
-        from_std_string_jstring(String::from(attribute.name.to_str()), env)
-    });
-    match result {
-        Ok(jstring) => jstring,
-        Err(err) => {
-            let exception_class = clone.find_class("java/lang/RuntimeException").unwrap();
-            let message = format!("Rust panic occurred in get_attribute_name: {:?}", err);
-            clone.throw_new(exception_class, message).unwrap();
-            clone.new_string("").unwrap().into()
-        }
+    let this = this as *mut Attribute;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Attribute::get_name called with null pointer",
+        )
+        .unwrap();
+
+        JObject::null().into()
+    } else {
+        let attr = &*(this as *mut Attribute);
+        env.new_string(attr.name.to_str())
+            .unwrap_or_else(|_| JObject::null().into())
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Attribute_get_1namespace<'local>(
-    env: JNIEnv<'local>,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Attribute_get_1namespace<
+    'local,
+>(
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    attr: jlong,
+    // non-null raw pointer to Attribute
```

Comment:
The `non-null` can't be true or there would be no need for the `this.is_null()` check below.
```suggestion
    // raw pointer to Attribute
```